src/app/basicauction/components/searchbar.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+
+import { useSearchParams, useRouter } from 'next/navigation';
+
+import DeleteIcon from '@/assets/svg/delete.svg';
+import SearchIcon from '@/assets/svg/search.svg';
+
+import * as S from './Searchbar.css';
+
+export default function SearchBar() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const [search, setSearch] = useState('');
+
+  const q = searchParams.get('q');
+
+  useEffect(() => {
+    setSearch(q || '');
+  }, [q]);
```

Comment:
```suggestion
  const q = searchParams.get('q');
  const [search, setSearch] = useState(q || '');

```

이렇게 진행하는 것은 어떤가요? 맨 처음 렌더링 될 때 쿼리 스트링에 있는 값을 가져와 state를 넣어줍니다.
이렇게 된다면 useEffect를 없앨 수 있는 것 같아요!